### PR TITLE
Use Cargo to build and run tests and benchmarks

### LINKED ISSUE #31: Use Cargo for tests, examples, benchmarks

### PR DESCRIPTION
**Pull request by aroben**
`html5ever-external-test` and `html5ever-external-bench` are now gone. Their custom test runners have moved into the individual test/bench files. We're using the `#[start]` attribute to define our own entry point (`#[main]` isn't respected for tests) which generates the test cases and calls through to `test::test_main`.

`tests/util.rs` now lives in its own `test_util` crate. It's the only way I could figure out to share this functionality between the tokenizer and tree builder integration tests.

`Makefile.in` is now a little simpler because we can just use `cargo test` to handle building the tests and examples and and running the tests, and `cargo bench` for building and running the benches. Top-level make targets like `build`, `check`, and `examples` continue to work (though `make examples` also builds the tests now).

Fixes #31.

### CODE REVIEW

**Comment by kmcallister** [comment]
Needs a rebase.

**Comment by aroben** [comment]
Updated and passing CI.

**Comment by aroben** [comment]
Just updated to also make `cargo bench` work.

**Comment by kmcallister** [comment]
Needs a rebase.

**Comment by aroben** [comment]
Rebased!

File: Makefile.in [hunk 1]
```diff
@@ -44,48 +36,25 @@ for_c: libhtml5ever_for_c.a
...
-bench: html5ever-external-bench
-	./html5ever-external-bench --bench
+bench:
+	(cd $(VPATH) && cargo bench)
 
 .PHONY: clean
 clean:
 	(cd $(VPATH) && cargo clean)
-	rm -f *.o *.a *.so *.dylib *.dll *.dummy *-test *-bench $(EXAMPLES)
+	rm -f *.o *.a *.so *.dylib *.dll *.dummy $(EXAMPLES)
```
Comment (kmcallister) [inline Makefile.in]: `$(EXAMPLES)` seems to be an undefined variable now.

Comment (aroben) [inline Makefile.in]: Fixed! Though of course now we need to rustup again :unamused:

**Comment by aroben** [comment]
Rebased once again.

**Comment by aroben** [comment]
But now `phf` is failing to build on the latest rustc. We'll get this PR merged someday!

**Comment by kmcallister** [comment]
Yep, I just landed a patch to rustc that breaks all use of compiler plugins ^_^

**Comment by aroben** [comment]
Rebased again.

**Comment by kmcallister** [comment]
Thanks!

**Comment by aroben** [comment]
Yay!

**Pull request merged**